cpp/src/arrow/compute/kernels/vector_run_end_encode.cc

```diff
@@ -563,6 +568,11 @@ void RegisterVectorRunEndEncode(FunctionRegistry* registry) {
   for (const auto& ty : NumericTypes()) {
     add_kernel(ty->id());
   }
+
+  // TODO: Adding HALF_FLOAT to FloatingPointTypes() causes cascading failures
+  // so we specify it manually here. We should add it to FloatingPointTypes()
+  add_kernel(Type::HALF_FLOAT);
+
```

Comment:
If any reviewers want to take a look at adding HALF_FLOAT to FloatingPointTypes I'd appreciate that. I made some initial progress fixing the crashes adding HALF_FLOAT to FloatingPointTypes causes but got stuck on some fairly obscure C++ templating errors.